git/git.go

```diff
@@ -63,16 +63,21 @@ func Commands(cmds ...*Cmd) (*State, error) {
 		return &State{}, err
 	}
 	s := &State{Dir: tdir}
+	err = CommandsWithState(s, cmds...)
+	return s, err
+}
+
+func CommandsWithState(s *State, cmds ...*Cmd) error {
```

Comment:
Because git.Commands returns error and With is called with a state as the receiver type. Previously, you couldn't batch commands with an already existing state.

Reply:
Hmm, I see. That is sort of a weakness in the existing API :thinking:

Reply:
How about we change the API to allow `commands` to take `With`.

Then every `Commands` call will have to be explicitly called with `Run()`. 

Prettier I think :thought_balloon:

Reply:
100% agree but that would mean to modify every file that uses the git package. U think its a good idea to bundle that with this PR or should i make another one.